wlutil/wlutil.py

```diff
@@ -71,6 +74,20 @@
 # Useful for defining lists of files (e.g. 'files' part of config)
 FileSpec = collections.namedtuple('FileSpec', [ 'src', 'dst' ])
 
+class SubmoduleError(Exception):
+    """Error representing a nonexistent or uninitialized submodule"""
+    def __init__(self, path):
+        self.message = "Dependency missing or not initialized " + \
+                str(path) + \
+                ". Do you need to initialize submodules?"
+        self.path = path
+
+    def __repr__(self):
+        return "Submodule Error: \n" + self.message
+
+    def __str__(self):
```

Comment:
stack overflow makes it sound like __str__ should default to __repr__, but that wasn't working for me. I'm not sure what's going on with that.